experiments/explanation/summarization.py

```diff
@@ -86,7 +86,8 @@ def get_explanation(model, text_input):
                                     verbose= True, 
                                     llm=ChatOpenAI(temperature=0.6, model= model, streaming=True, callbacks=[MyCustomSyncHandler()]),)
     bit_response = conversation.predict(input=text_input)
-
+    if bit_response[-1] != "." or bit_response[-1] != "!" or bit_response[-1] != "?":
+        bit_response = conversation.predict(input="continue")
```

Comment:
I guess the  bit_response can be any of (.!?) when the prediction reaches max_tokens? is there another message OpenAI sends when this happens? like a warning or error.

Reply:
I am not getting any error or warning but I'll try to find a better method of detecting whether the message is complete or cut off